x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockAccountingTests.java

```diff
@@ -130,13 +130,11 @@ public void testBooleanBlock() {
         Block emptyPlusOne = new BooleanArrayBlock(new boolean[] { randomBoolean() }, 1, new int[] { 0 }, null, Block.MvOrdering.UNORDERED);
         assertThat(emptyPlusOne.ramBytesUsed(), is(alignObjectSize(empty.ramBytesUsed() + 1) + alignObjectSize(Integer.BYTES)));
 
-        boolean[] randomData = new boolean[randomIntBetween(1, 1024)];
-        int[] valueIndices = IntStream.range(0, randomData.length).toArray();
+        boolean[] randomData = new boolean[randomIntBetween(2, 1024)];
+        int[] valueIndices = IntStream.range(0, randomData.length + 1).toArray();
```

Comment:
We're building block arrays directly in the test (rather than using a builder), and must construct a valid structured block so that we can filter on it. A block with firstValuesIndex must have a final element entry in to hold the end of the final position. This is why we +1 here - each position entry has just one entry, and the last has the final end position.  Note: builders do this for you, but we want to be more specific in this test. A separate test scenario verifies filtering on the final position.